File: js/test/ladder.test.js
/*
 * Ladder Game Tests - Data-driven from YAML
 */

"use strict";

import { test } from 'vitest';
import { loadTest } from './yamlTestLoader';
import { executeTest } from './yamlTestExecutor';

test('standingStillShowsGSymbol', async () => {
	await executeTest(loadTest('standing-still-shows-g-symbol.yaml'));
});

test('moveRightAndContinueUntilWall', async () => {
	await executeTest(loadTest('move-right-and-continue-until-wall.yaml'));
});

test('moveLeftAndContinueUntilWall', async () => {
	await executeTest(loadTest('move-left-and-continue-until-wall.yaml'));
});

test('basicFall', async () => {
	await executeTest(loadTest('basic-fall.yaml'));
});

test('jumpInPlaceAllFrames', async () => {
	await executeTest(loadTest('jump-in-place-all-frames.yaml'));
});

test('jumpRightAllFrames', async () => {
	await executeTest(loadTest('jump-right-all-frames.yaml'));
});

test('jumpRightSimultaneous', async () => {
	await executeTest(loadTest('jump-right-simultaneous.yaml'));
});

test('jumpLeftAllFrames', async () => {
	await executeTest(loadTest('jump-left-all-frames.yaml'));
});

test('jumpRightLandOnLadder', async () => {
	await executeTest(loadTest('jump-right-land-on-ladder.yaml'));
});

test('moveRightAndClimbLadder', async () => {
	await executeTest(loadTest('move-right-and-climb-ladder.yaml'));
});

test('jumpClimbsLadder', async () => {
	await executeTest(loadTest('jump-climbs-ladder.yaml'));
});

test('upClimbsLadder', async () => {
	await executeTest(loadTest('up-climbs-ladder.yaml'));
});

test('ladMovesThroughLadder', async () => {
	await executeTest(loadTest('lad-moves-through-ladder.yaml'));
});

test('reachingGoalEndsLevel', async () => {
	await executeTest(loadTest('reaching-goal-ends-level.yaml'));
});

test('hittingSpikeEndsGame', async () => {
	await executeTest(loadTest('hitting-spike-ends-game.yaml'));
});

test('hittingBarrelEndsGame', async () => {
	await executeTest(loadTest('hitting-barrel-ends-game.yaml'));
});

test('asteriskCollectsBarrels', async () => {
	await executeTest(loadTest('asterisk-collects-barrels.yaml'));
});

test('ladDeathOnAsterisk', async () => {
	await executeTest(loadTest('lad-death-on-asterisk.yaml'));
});

test('ladDeathOnProducer', async () => {
	await executeTest(loadTest('lad-death-on-producer.yaml'));
});

test('barrelsFindLadder', async () => {
	await executeTest(loadTest('barrels-find-ladder.yaml'));
});

test('collectStatueIncreasesScore', async () => {
	await executeTest(loadTest('collect-statue-increases-score.yaml'));
});

test('cyclesDecrementEachFrame', async () => {
	await executeTest(loadTest('cycles-decrement-each-frame.yaml'));
});

test('timeRunningOutEndsGame', async () => {
	await executeTest(loadTest('time-running-out-ends-game.yaml'));
});

test('disappearingFloorTest', async () => {
	await executeTest(loadTest('disappearing-floor-test.yaml'));
});

File: java/test/java/com/Ostermiller/Ladder/LadderTest.java
/*
 * Part of Ladder, a game.
 * Copyright (C) 2026
 * Stephen Ostermiller http://ostermiller.org/contact.pl?regarding=Ladder
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * See COPYING.TXT for details.
 */

package com.Ostermiller.Ladder;

import org.junit.Test;

public class LadderTest {

	@Test
	public void standingStillShowsGSymbol() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("standing-still-shows-g-symbol.yaml")
		);
	}

	@Test
	public void moveRightAndContinueUntilWall() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("move-right-and-continue-until-wall.yaml")
		);
	}

	@Test
	public void moveLeftAndContinueUntilWall() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("move-left-and-continue-until-wall.yaml")
		);
	}

	@Test
	public void basicFall() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("basic-fall.yaml")
		);
	}

	@Test
	public void jumpInPlaceAllFrames() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("jump-in-place-all-frames.yaml")
		);
	}

	@Test
	public void jumpRightAllFrames() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("jump-right-all-frames.yaml")
		);
	}

	@Test
	public void jumpRightSimultaneous() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("jump-right-simultaneous.yaml")
		);
	}

	@Test
	public void jumpLeftAllFrames() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("jump-left-all-frames.yaml")
		);
	}

	@Test
	public void jumpRightLandOnLadder() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("jump-right-land-on-ladder.yaml")
		);
	}

	@Test
	public void moveRightAndClimbLadder() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("move-right-and-climb-ladder.yaml")
		);
	}

	@Test
	public void jumpClimbsLadder() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("jump-climbs-ladder.yaml")
		);
	}

	@Test
	public void upClimbsLadder() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("up-climbs-ladder.yaml")
		);
	}

	@Test
	public void ladMovesThroughLadder() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("lad-moves-through-ladder.yaml")
		);
	}

	@Test
	public void reachingGoalEndsLevel() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("reaching-goal-ends-level.yaml")
		);
	}

	@Test
	public void hittingSpikeEndsGame() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("hitting-spike-ends-game.yaml")
		);
	}

	@Test
	public void hittingBarrelEndsGame() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("hitting-barrel-ends-game.yaml")
		);
	}

	@Test
	public void asteriskCollectsBarrels() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("asterisk-collects-barrels.yaml")
		);
	}

	@Test
	public void ladDeathOnAsterisk() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("lad-death-on-asterisk.yaml")
		);
	}

	@Test
	public void ladDeathOnProducer() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("lad-death-on-producer.yaml")
		);
	}

	@Test
	public void barrelsFindLadder() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("barrels-find-ladder.yaml")
		);
	}

	@Test
	public void collectStatueIncreasesScore() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("collect-statue-increases-score.yaml")
		);
	}

	@Test
	public void cyclesDecrementEachFrame() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("cycles-decrement-each-frame.yaml")
		);
	}

	@Test
	public void timeRunningOutEndsGame() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("time-running-out-ends-game.yaml")
		);
	}

	@Test
	public void disappearingFloorTest() throws Exception {
		YamlTestExecutor.executeTest(
			YamlTestLoader.loadTest("disappearing-floor-test.yaml")
		);
	}
}
